packages/tenant-process/src/services/readModelService.ts

```diff
@@ -366,6 +368,98 @@ export function readModelServiceBuilder(config: TenantProcessConfig) {
         };
       }
     },
+
+    async getCertifiedAttributes({
+      certifierId,
+      offset,
+      limit,
+    }: {
+      certifierId: string;
+      offset: number;
+      limit: number;
+    }): Promise<ListResult<ApiCertifiedAttribute>> {
+      const aggregationPipeline: Document[] = [
+        {
+          $match: {
+            "data.kind": attributeKind.certified,
+            "data.origin": certifierId,
+          },
+        },
+        {
+          $lookup: {
+            from: "tenants",
+            localField: "data.id",
+            foreignField: "data.attributes.id",
+            as: "tenants",
+          },
+        },
+        { $unwind: "$tenants" },
+        { $unwind: "$tenants.data.attributes" },
+        {
+          $addFields: {
+            notRevoked: {
+              $cond: {
+                if: {
+                  $and: [
+                    { $eq: ["$tenants.data.attributes.id", "$data.id"] },
+                    { $not: ["$tenants.data.attributes.revocationTimestamp"] },
+                  ],
+                },
+                then: true,
+                else: false,
+              },
+            },
+          },
+        },
+        {
+          $match: {
+            notRevoked: true,
+          },
+        },
+        {
+          $project: {
+            _id: 0,
+            id: "$tenants.data.id",
+            name: "$tenants.data.name",
+            attributeId: "$data.id",
+            attributeName: "$data.name",
+            lowerName: { $toLower: "$tenants.data.name" },
+          },
+        },
+        {
+          $sort: {
+            lowerName: 1,
+          },
+        },
+      ];
+
+      const data = await attributes
+        .aggregate(
+          [...aggregationPipeline, { $skip: offset }, { $limit: limit }],
+          { allowDiskUse: true }
+        )
+        .toArray();
+
+      const result = z.array(ApiCertifiedAttribute).safeParse(data);
```

Comment:
`readmodelservice` should never return an `Api-object`. It should only return the object itself, parsed according to its definition. The parsing from object to `Api-object` should only happen at router level.

Reply:
This is not possible in this case, since there is no parsing from an object model to another. The mongo query returns the shape of the Api-object itself. The only "solution" to make it work that way is to make a type that is the exact copy of the Api-object, return that type from the `readModelService` method and then cast it to Api-object at route level. Do you have any other idea?

Reply:
In many cases the parsing from `object` to `ApiObject` is almost 1:1. I think we should follow the pattern here as well. @galales what do you think?

Reply:
both points are actually valid, we should keep the two data layer separated, but the query is currently tailored on the api response.
I would suggest to create an internal type like 
```
type TheNewQueryReturnType = z.infer<ApiCertifiedAttribute>;
```
that provides
- logical model separation
- clarity about the current equality of the two models
- the possibility to tune definitions if we would need to distinguish the two models

What do you think?

Reply:
I didn't think as the readModelService and the router as two separated layers, make sense!